### PR TITLE
Add class skeletons for imperative workflow control

### PR DESCRIPTION
**Pull request by dbmikus**
This is how I'm thinking about the imperative interface for managing a workflow:

- transitioning between tasks and steps
- adding documents and forms to the workflow

### CODE REVIEW

File: src/fixpoint/analyze/memory.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ class _DataDict(TypedDict):
     all_tool_call_names: List[List[str]]
     all_tool_call_args: List[List[str]]
 
-    workflow_id: List[Optional[str]]
+    workflow_name: List[Optional[str]]
```
Comment (jdc123) [inline src/fixpoint/analyze/memory.py:34]: Why the change from `workflow_id` to `workflow_name`?

File: src/fixpoint/workflow/_workflow.py [hunk 2]
```diff
@@ -11,9 +11,13 @@
 class Workflow(SupportsWorkflow):
```
Comment (jdc123) [inline src/fixpoint/workflow/_workflow.py:11]: I think there are certain things that we should add here that are inherent to the Workflow:
1. Tasks - I think a workflow should have tasks that you can add / remove.
2. Versioning - Behind the scenes whenever you add a task object the workflow version changes. 
3. Runs - I think a workflow object should start / stop runs

File: src/fixpoint/workflow/_workflow.py [hunk 3]
```diff
@@ -11,9 +11,13 @@
...
     """A simple workflow implementation"""
 
-    id: str
+    name: str
     display_name: Optional[str]
+    run_id: str
```
Comment (jdc123) [inline src/fixpoint/workflow/_workflow.py:16]: I think a workflow object should not be tied to a single run. We should come up with a separate abstraction for it which we could call WorkflowRun. The run could then have a run_id and a workflow_id

File: src/fixpoint/analyze/memory.py (hunk 1 above)
Comment (dbmikus) [reply src/fixpoint/analyze/memory.py:34]: IDs are system-generated and immutable. They are like a file descriptor. Names can be provided by a user, and they are like a file name.

I am wondering if this distinction will be confusing. For an advanced API, I think it makes sense to have both an unchangeable identifier and a unique human-readable name. Not sure if the difference will trip up users.

File: src/fixpoint/workflow/_workflow.py (hunk 2 above)
Comment (dbmikus) [reply src/fixpoint/workflow/_workflow.py:11]: Per our convo:

1. you don't explicitly add tasks, you just goto them and that creates the task implicitly
2. versioning: I agree, but I don't think this object really needs to care about that
3. I am going to rename this a `WorkflowRun` and for now we will not have a `Workflow` class

**Review by jdc123 — APPROVED** [review]
